src/main/java/CharArraySolution.java

```diff
@@ -1,44 +1,41 @@
 import java.util.Arrays;
 
 public class CharArraySolution {
-    int string_counter = 0;
 
     public int solveEquation(String equation) {
 
-        if (Utils.getInstance().verifyEquationInput(equation)) {
+        if (Utils.verifyEquationInput(equation)) {
             char[] characters = equation.toCharArray();
-            int operatorLocation = findOperator(characters);
+            int operator_location = findOperator(characters);
 
-            if (verifyOperator(Character.toString(characters[operatorLocation]), operatorLocation)) {
-                int numBefore = Integer.parseInt(
-                        String.valueOf(
-                                Arrays.copyOfRange(characters, 0, operatorLocation))
-                );
-                int numAfter = Integer.parseInt(
-                        String.valueOf(
-                                Arrays.copyOfRange(characters, operatorLocation + 1, characters.length))
-                );
-                return Utils.getInstance().calcRes(Character.toString(characters[operatorLocation]), numBefore, numAfter);
+            if (operator_location != -1) {
```

Comment:
To avoid complexity  we can do:

```
if (operator_location == -1) {
   return operator_location;
}
```
 In this case we do not need to write `else` statement